packages/react/ds-app-launchpad/src/ui/MarkdownEditor/hooks/useEditor.ts

```diff
@@ -0,0 +1,276 @@
+import { useCallback, useEffect } from "react";
+
+type ToolbarAction = {
+  label: string;
+  shortcut: string | undefined;
+  handler: () => void;
+};
+
+type EditorActions = {
+  italic: ToolbarAction;
+  bold: ToolbarAction;
+  title: ToolbarAction;
+  quote: ToolbarAction;
+  code: ToolbarAction;
+  uList: ToolbarAction;
+  oList: ToolbarAction;
+  link: ToolbarAction;
+};
+
+const useEditor = (
+  textareaRef: React.RefObject<HTMLTextAreaElement | null>,
+) => {
+  const wrapText = useCallback(
+    (character: string) => {
+      const textarea = textareaRef.current;
+      if (!textarea) return;
+
+      const { selectionStart, selectionEnd } = textarea;
+      const isSelected = selectionStart !== selectionEnd;
+      textarea.focus();
+
+      if (isSelected) {
+        const selectedText = textarea.value.slice(selectionStart, selectionEnd);
+        const wrappedText = character + selectedText + character;
+        document.execCommand("insertText", false, wrappedText);
```

Comment:
`execCommand` is [now obsolete](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand). I'd recommend against using it.

Reply:
I haven't found an alternative more modern way of doing this.

Reply:
Would using a small reusable function like this work?
```typescript
const pasteText = (textArea: HTMLTextAreaElement, inputText: string, inputMode: SelectionMode = "end") => {
  textArea.setRangeText(inputText, textArea.selectionStart, textArea.selectionEnd, inputMode);
};
// usage, in prefixLines(), for example...
// ....
textarea.setSelectionRange(startPos, endPos);
pasteText(textarea, wrappedLines.join("\n"));
```

Reply:
setRangeText seems to be working for inserting text however the undo functionality is not working as expected.
I think it is ok to use `execCommand` for this usecase since there is 100% alternative in browsers.

Reply:
Thanks for exploring, I'm happy to leave this in then